code/modules/mob/living/living.dm

```diff
@@ -382,12 +382,29 @@ default behaviour is:
 	//Getting out of someone's inventory.
 	if(istype(src.loc, /obj/item/weapon/holder))
 		escape_inventory(src.loc)
-		return
+		return TRUE
 
 	//unbuckling yourself
-	if(buckled && !HAS_TRAIT(src, TRAIT_BUCKLED))
-		spawn() escape_buckle()
-		return TRUE
+	if(buckled)
+		if(HAS_TRAIT(src, TRAIT_BUCKLED))
+			var/list/sources = status_traits[TRAIT_BUCKLED]
+			for(var/i in sources)
+				if(istype(i, /obj/item/javelin))
```

Comment:
aghh noo this is not what i said. Don't put specific item paths in here

resist_buckle is already called on the buckled item, just override that in the javelin, and redirect attack_hand to it too